components/org.wso2.carbon.identity.conditional.auth.functions.user/src/test/java/org/wso2/carbon/identity/conditional/auth/functions/user/IsMemberOfAnyOfGroupsFunctionImplTest.java

```diff
@@ -0,0 +1,112 @@
+import org.wso2.carbon.identity.common.testng.WithCarbonHome;
+import org.wso2.carbon.identity.common.testng.WithH2Database;
+import org.wso2.carbon.identity.common.testng.WithRealmService;
+import org.wso2.carbon.identity.conditional.auth.functions.test.utils.sequence.JsSequenceHandlerAbstractTest;
+import org.wso2.carbon.identity.conditional.auth.functions.user.internal.UserFunctionsServiceHolder;
+import org.wso2.carbon.identity.core.util.IdentityTenantUtil;
+import org.wso2.carbon.user.api.UserRealm;
+import org.wso2.carbon.user.core.service.RealmService;
+
+import java.util.Collections;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+
+@WithCarbonHome
+@WithH2Database(files = "dbscripts/h2.sql")
+@WithRealmService(injectToSingletons = {UserFunctionsServiceHolder.class, IdentityTenantUtil.class,
+        FrameworkServiceDataHolder.class})
+public class IsMemberOfAnyOfGroupsFunctionImplTest extends JsSequenceHandlerAbstractTest {
+
+    @WithRealmService
+    private RealmService realmService;
+
+    @BeforeMethod
+    protected void setUp() throws Exception {
+
+        super.setUp();
+        sequenceHandlerRunner.registerJsFunction("isMemberOfAnyOfGroups",
+                new IsMemberOfAnyOfGroupsFunctionImpl());
+        UserRealm userRealm = realmService.getTenantUserRealm(-1234);
+        userRealm.getUserStoreManager().addRole("students",
+                new String[]{"user1", "user2"}, null);
+        userRealm.getUserStoreManager().addRole("teachers",
+                new String[]{"user1", "user3"}, null);
+    }
+
+    @AfterMethod
+    public void tearDown() {
+
+    }
+
+    @Test(dataProvider = "isMemberOfAnyOfGroupsDataProvider")
+    public void testHasAnyOfTheRoles(String user, boolean steppedUp) throws Exception {
+
+        sequenceHandlerRunner.addSubjectAuthenticator("BasicMockAuthenticator", user, Collections.emptyMap());
+
+        ServiceProvider sp1 = sequenceHandlerRunner.loadServiceProviderFromResource("isMemberOfAnyOfGroups-test-sp.xml",
+                this);
+
+        AuthenticationContext context = sequenceHandlerRunner.createAuthenticationContext(sp1);
+
+        SequenceConfig sequenceConfig = sequenceHandlerRunner
+                .getSequenceConfig(context, sp1);
+        context.setSequenceConfig(sequenceConfig);
+        context.initializeAnalyticsData();
+
+        HttpServletRequest req = sequenceHandlerRunner.createHttpServletRequest();
+        HttpServletResponse resp = sequenceHandlerRunner.createHttpServletResponse();
+
+        sequenceHandlerRunner.handle(req, resp, context, "carbon.super");
+
+        HttpServletRequest secondReq = sequenceHandlerRunner.createHttpServletRequest();
+        secondReq.setAttribute("s", "S");
+        HttpServletResponse secondResp = sequenceHandlerRunner.createHttpServletResponse();
```

Comment:
Doesn't seem to be used. Let's remove this block

Reply:
Fixed